src/blocks/content-helper/content-helper-provider.ts

```
@@ -0,0 +1,118 @@
+
+		// Get post's author.
+		const currentPost = editor.getCurrentPost() as Post;
+		const author = select( 'core' ).getEntityRecord( 'root', 'user', currentPost.author ) as User;
+
+		// Get post's first category.
+		const categoryId = editor.getEditedPostAttribute( 'categories' ) as Array<number>[0];
+		const category = select( 'core' ).getEntityRecord( 'taxonomy', 'category', categoryId ) as Taxonomy;
+
+		// Get post's first tag.
+		const tagId = editor.getEditedPostAttribute( 'tags' ) as Array<number>[0];
+		const tag = select( 'core' ).getEntityRecord( 'taxonomy', 'post_tag', tagId ) as Taxonomy;
+
+		// Fetch data from the API.
+		const fetchQueryResult = this.buildFetchDataQuery( author, category, tag );
+		if ( fetchQueryResult.query === null ) {
+			return Promise.reject( fetchQueryResult.message );
+		}
+		const data = await this.fetchData( fetchQueryResult );
+
+		// Set the content helper's message.
+		let message = `${ __( 'Top-performing posts', 'wp-parsely' ) } ${ fetchQueryResult.message }.`;
+		if ( data.length === 0 ) {
+			message = `${ __( 'The Parse.ly API did not return any results for top-performing posts', 'wp-parsely' ) } ${ fetchQueryResult.message }.`;
+		}
+
+		return { message, posts: this.processData( data ) };
+	}
+
+	private static async fetchData( fetchDataQueryResult: BuildFetchDataQueryResult ): Promise<SuggestedPost[]> {
+		let response;
+		let error;
+
+		try {
+			response = await apiFetch( {
+				path: addQueryArgs( '/wp-parsely/v1/analytics/posts', fetchDataQueryResult.query ),
+			} ) as ApiResponse;
+		} catch ( wpError ) {
+			error = wpError;
+		}
+
+		if ( response?.error ) {
+			error = response.error;
+		}
+
+		if ( error ) {
+			return Promise.reject( error );
+		}
+
+		return response?.data || [];
+	}
+
+	private static processData( data: SuggestedPost[] ): SuggestedPost[] {
+		return data.map( ( p ) => {
+			// @ts-ignore
+			p.statsUrl = `${ window.wpParselyContentHelperPrefix }?url=${ p.url }`;
+			return p;
```

- What error is being ignored?
- Encode the URL

```suggestion
			const statsUrl = `${ window.wpParselyContentHelperPrefix }?url=${ window.encodeURIComponent( p.url ) }`;
			return Object.assign( p, { statsUrl } );
```

---

We're getting a "Property 'wpParselyContentHelperPrefix' does not exist on type 'Window & typeof globalThis" error.

---

Ideally we could import types provided by Parse.ly's JavaScript library. If those don't exist, we can declare our own (alongside others, like for SVG imports).

https://www.jamestharpe.com/typescript-extend-window/